sockeye/encoder.py

```diff
@@ -74,20 +79,24 @@ def get_recurrent_encoder(config: RecurrentEncoderConfig, fused: bool,
     :return: Encoder instance.
     """
     # TODO give more control on encoder architecture
-    encoders = list()
+    encoders = list()  # type: List[Encoder]
 
     encoders.append(Embedding(num_embed=config.num_embed,
                               vocab_size=config.vocab_size,
                               prefix=C.SOURCE_EMBEDDING_PREFIX,
                               dropout=config.embed_dropout,
                               embed_weight=embed_weight))
+
+    if config.reverse_input:
```

Comment:
would it make sense to run this after BatchMajor2TimeMajor in case the convolutional encoder is not present in order to save the additional swapaxes calls?